apple/RNCWebView.m

```diff
@@ -309,8 +310,18 @@ - (BOOL)canPerformAction:(SEL)action withSender:(id)sender
  */
 - (WKWebView *)webView:(WKWebView *)webView createWebViewWithConfiguration:(WKWebViewConfiguration *)configuration forNavigationAction:(WKNavigationAction *)navigationAction windowFeatures:(WKWindowFeatures *)windowFeatures
 {
-  if (!navigationAction.targetFrame.isMainFrame) {
-    [webView loadRequest:navigationAction.request];
+  if (!navigationAction.targetFrame) {
```

Comment:
I'm not sure about how objective-c handle null values, but based on this documentation it should always be `nil` : https://developer.apple.com/documentation/webkit/wknavigationaction/1401918-targetframe

If just made some checks, the following code would print `targetFrame is nill`
```c
if (!navigationAction.targetFrame) {
  RCTLogWarn(@"targetFrame is nill");
} else if (!navigationAction.targetFrame.isMainFrame) {
  RCTLogWarn(@"isMainFrame is nill");
} 
```

And the following one would print `isMainFrame is nill`
```c
if (!navigationAction.targetFrame.isMainFrame) {
  RCTLogWarn(@"isMainFrame is nill");
} else if (!navigationAction.targetFrame) {
  RCTLogWarn(@"targetFrame is nill");
}
```

Seems like objective-c has natural "optional chaining" like mechanism? 🤔 

However the [original commit](https://github.com/react-native-webview/react-native-webview/commit/6643a8e38d12df8a857ae09839c94ef158e14176) links to this stackoverflow answer: https://stackoverflow.com/a/25853806 that uses `navigationAction.targetFrame.isMainFrame`
But there is another answer https://stackoverflow.com/a/27391215 that reacts to the first one and uses `navigationAction.targetFrame == nil` instead.

So I suppose my new version would be enough?

However my implementation had a useless check in the `else` block (already covered by the surrounding `if`), so I removed it here: https://github.com/react-native-webview/react-native-webview/compare/3ba2f992c1a6349e35796a50f13bcdf6ba80df60..6fe3b597c9107a88855da7b523f3970449ea3adb

Reply:
> Seems like objective-c has natural "optional chaining" like mechanism?

Huh! It seems you are correct. TIL. https://collindonnell.com/2022/04/28/the-brilliance-of-nil-in-objective-c/

> my implementation had a useless check in the else block (already covered by the surrounding if)

Are you _sure_ that was useless? Given that the previous logic checked `isMainFrame` and your surrounding `if` just checks `targetFrame`, wouldn't keeping that `isMainFrame` check in the `else` be an appropriate way to preserve the previous behavior?

And yes I see the other StackOverflow answer that uses `targetFrame == nil` instead of `isMainFrame`, but since that doesn't explain _why_ it makes that change, I'd be inclined to leave the existing logic just to be safe...

This reminds me of another question: your examples in the description use `window.open('https://example.com', '_blank')`. Is the behavior any different if the second parameter isn't `_blank`? (would `targetFrame` still be `nil`?)

Reply:
> Are you sure that was useless? Given that the previous logic checked isMainFrame and your surrounding if just checks targetFrame, wouldn't keeping that isMainFrame check in the else be an appropriate way to preserve the previous behavior?

I would say yes. This code (original PR):
```c
  if (!navigationAction.targetFrame) {
    NSURL *url = navigationAction.request.URL;
    if (_onOpenWindow) {
      NSMutableDictionary<NSString *, id> *event = [self baseEvent];
      [event addEntriesFromDictionary: @{@"targetUrl": url.absoluteString}];
      _onOpenWindow(event);
    } else {
      if (!navigationAction.targetFrame.isMainFrame) {
        [webView loadRequest:navigationAction.request];
      }
    }
  }
```

Can be simplified as
```c
  if (!navigationAction.targetFrame) {
    //...
      if (!navigationAction.targetFrame.isMainFrame) {
        [webView loadRequest:navigationAction.request];
      }
  }
```

If we enter the first `if`, then it should be impossible to have something in `targetFrame` in the second `if`. So 2nd `if` should alway be `true`

> Huh! It seems you are correct. TIL. https://collindonnell.com/2022/04/28/the-brilliance-of-nil-in-objective-c/

> but since that doesn't explain why it makes that change, I'd be inclined to leave the existing logic just to be safe...

As your link confirm the `nil` chaining mechanism, I see no cons to add `isMainFrame` back. So lets do it: https://github.com/react-native-webview/react-native-webview/compare/6fe3b597c9107a88855da7b523f3970449ea3adb..e82aed74391b74611b143ccc73cd06d53aebcfde

> This reminds me of another question: your examples in the description use window.open('https://example.com', '_blank'). Is the behavior any different if the second parameter isn't _blank? (would targetFrame still be nil?)

Good question. I completed the tests to check that : https://github.com/react-native-webview/react-native-webview/compare/e82aed74391b74611b143ccc73cd06d53aebcfde..38ba351763ad6ddfab3b002bc227d2d0c19a233d

Result on iOS is:
- no param -> trigger OnNewWindow
- '_blank' -> trigger OnNewWindow
- '_self' -> navigate in same WebView
- '_parent' -> navigate in same WebView
- '_top' -> navigate in same WebView

I did not test yet on Android, I'll try to do it later today.

Reply:
I'm made the test on Android, same behaviour 👍

Reply:
> If we enter the first if, then it should be impossible to have something in targetFrame in the second if.

🤦 
Of course. Sorry.

> Result on iOS is:
> no param -> trigger OnNewWindow
'_blank' -> trigger OnNewWindow
'_self' -> navigate in same WebView
'_parent' -> navigate in same WebView
'_top' -> navigate in same WebView

Thanks for checking. Is that the behavior you _want_? I'm not sure if those are special values that are _supposed_ refer to the current window, but I know that this parameter could instead be a name (e.g. "MySpecialPopup"), in which case a _browser_ would be expected to open the link in a new window, not navigate in the current one.

Reply:
Okay, I pulled your branch and ran my own test using a custom name, and it did trigger `onOpenWindow` as expected.

Reply:
Hi, sorry, I missed this question so I didn't realize that there were still some verifications to be done. Thank you for the test.